core/domain/skill_domain_test.py

```diff
@@ -114,6 +114,10 @@ def test_misconception_validation(self):
         self._assert_validation_error(
             'Expected misconception name to be a string')
 
+        self.skill.misconceptions[0].skill_id = 'incorrect_id'
```

Comment:
This to me seems a bit weird, that we have skill.misconception.skill_id...we can anyway pick up the id from skill.id, we dont need to keep it within misconception as well

Reply:
Reverted the changes.